src/nbl/asset/interchange/CGLTFLoader.h

```diff
@@ -0,0 +1,539 @@
+					std::optional<uint32_t> weights;				//!< The weights of the instantiated Morph Target. Number of elements must match number of Morph Targets of used mesh.
+
+					struct SGLTFNTransformationTRS
+					{
+						core::vector3df_SIMD translation;			//!< The node's translation along the x, y, and z axes.
+						core::vector3df_SIMD scale;					//!< The node's non-uniform scale, given as the scaling factors along the x, y, and z axes.
+						core::vector4df_SIMD rotation;				//!< The node's unit quaternion rotation in the order (x, y, z, w), where w is the scalar.
+					};
+
+					union SGLTFTransformation
+					{
+						SGLTFTransformation() {}
+						~SGLTFTransformation() {}
+
+						SGLTFTransformation(SGLTFTransformation& copy)
+						{
+							std::memmove(this, &copy, sizeof(SGLTFTransformation));
+						}
+
+						SGLTFTransformation(const SGLTFTransformation& copy)
+						{
+							std::memmove(this, &copy, sizeof(SGLTFTransformation));
+						}
+
+						SGLTFTransformation& operator=(SGLTFTransformation& copy)
+						{
+							std::memmove(this, &copy, sizeof(SGLTFTransformation));
+							return *this;
+						}
+
+						SGLTFTransformation& operator=(const SGLTFTransformation& copy)
+						{
+							std::memmove(this, &copy, sizeof(SGLTFTransformation));
+							return *this;
+						}
+
+						SGLTFNTransformationTRS trs;	
+						core::matrix4SIMD matrix;		//!< A floating-point 4x4 transformation matrix stored in column-major order.
+					} transformation;
+
+					/*
+						A set of primitives to be rendered. A node can contain one mesh.
+						A node's transform places the mesh in the scene.
+					*/
+
+					struct SGLTFMesh
+					{
+						struct SPrimitive
+						{
+							std::optional<uint32_t> indices;									//!< The index of the accessor that contains the indices.
+							std::optional<uint32_t> material;									//!< The index of the material to apply to this primitive when rendering.
+							std::optional<uint32_t> mode;										//!< The type of primitives to render.
+							std::optional<std::unordered_map<std::string, uint32_t>> targets;	//!< An array of Morph Targets, each Morph Target is a dictionary mapping attributes (only POSITION, NORMAL, and TANGENT supported) to their deviations in the Morph Target.
+
+							/*
+								An accessor provides a typed view into a bufferView or a subset of a bufferView similar to
+								how WebGL's vertexAttribPointer() defines an attribute in a buffer.
+							*/
+
+							struct SGLTFAccessor
+							{
+								std::optional<uint32_t> bufferView;
+								std::optional<uint32_t> byteOffset;
+								std::optional<uint32_t> componentType;
+								std::optional<bool> normalized;
+								std::optional<uint32_t> count;
+								std::optional<std::string> type;
+								std::optional<std::vector<double>> max; // todo - common number types
+								std::optional<std::vector<double>> min; // todo - common number types
```

Comment:
vectorSIMDf 

you probably dont need to use doubles for this, otherwise leave the use of doubles to a TODO